pkg/credentials/gitcreds/creds_test.go

```diff
@@ -305,8 +307,8 @@ func TestSSHFlagHandlingThrice(t *testing.T) {
     HostName github.com
     IdentityFile %s
     Port 22
-Host gitlab.com
-    HostName gitlab.com
+Host github.com
```

Comment:
Does having multiple exact same `Host` works ? I read a bit `man ssh_config` and I am not sure it's documented (either works or not), but I have a fear it doesn't as it's gonna pick up the first one and if it fails, it will considered it failed.

I think it's possible to pass multiple `IdentityFile` though, like 

```
Host github.com
    IdentityFile /etc/ssh/my_project_1_github_deploy_key
    IdentityFile /etc/ssh/my_project_2_github_deploy_key
```

Reply:
I think you're right. Reading https://superuser.com/q/366649 it looks like the correct way to handle this is with multiple differently-named `Host`s which each specify the same `HostName`

I've updated the PR to do this, where each `Host` includes the secret name to dedupe.

Reply:
Seems I was wrong! Thanks e2e tests! :)

Reply:
:thinking: